src/voice/VoiceRecorder.ts

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as Recorder from 'opus-recorder';
+import encoderPath from 'opus-recorder/dist/encoderWorker.min.js';
+import {MatrixClient} from "matrix-js-sdk/src/client";
+import CallMediaHandler from "../CallMediaHandler";
+import {SimpleObservable} from "matrix-widget-api";
+
+export class VoiceRecorder {
+    private recorder = new Recorder({
+        encoderPath, // magic from webpack
+        mediaTrackConstraints: <MediaTrackConstraints>{
+            deviceId: CallMediaHandler.getAudioInput(),
+        },
+        encoderSampleRate: 16000, // we could go down to 12khz, but we lose quality
+        encoderApplication: 2048, // voice (default is "audio")
+        streamPages: true, // so we can have a live EQ for the user
+        encoderFrameSize: 10, // we want updates fairly regularly for the UI
```

Comment:
units? 20ms is fairly standard frame size for 'real time'

Reply:
ms indeed - let's try 20ms